### PR TITLE
 *: switch to dep for dependency management

### PR DESCRIPTION
**Pull request by ericchiang**

### CODE REVIEW

File: scripts/verify-gopkg.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -e
```
Comment (colemickens) [inline scripts/verify-gopkg.sh]: Did you notice anything that fails this check so far? Or is this an assert style sanity check?

This also doesn't seem to assert that the assets in the vendor/ are valid corresponding to either the toml/lock. Is that worth doing as well?

File: scripts/verify-gopkg.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+TOML=$( grep -P '( |^)name = ".*"' Gopkg.toml | tr -d ' ' | sort )
+LOCK=$( grep -P '( |^)name = ".*"' Gopkg.lock | tr -d ' ' | sort )
+
+# NOTE(ericchiang): For some reason you need this "export" here for bash reasons
+# I don't understand.
+export DIFF=$( diff <(echo "$TOML") <(echo "$LOCK") )
```
Comment (colemickens) [inline scripts/verify-gopkg.sh]: Given you comment, and a quick test, did you ensure this works in the failure case?

You don't wrap the value in quotes, and `DIFF=` and `export DIFF=` work differently if the values have spaces, at least in my quick little experiment it seemed to.

It seems like maybe `export` is papering over the fact that your value has whitespace and allowing for false positives?

Comment (rphillips) [reply scripts/verify-gopkg.sh]: might need a `-q` in there as well to only print when files differ

File: scripts/verify-gopkg.sh (hunk 1 above)
Comment (ericchiang) [reply scripts/verify-gopkg.sh]: Assert style sanity check.

File: scripts/verify-gopkg.sh (hunk 2 above)
Comment (ericchiang) [reply scripts/verify-gopkg.sh]: I confirmed that this works when the files are out of sync.

If I comment out the `export` and modify one of the Gopkg files I can replicate locally that `DIFF` becomes set to the empty string no matter what.

I can add some quotes, though I tested that and it didn't work to fix this problem. @thorfour and I were staring at this for a while and don't have any good answer here.

Comment (rphillips) [inline scripts/verify-gopkg.sh]: should be:

```
DIFF=$( diff <(echo "$TOML") <(echo "$LOCK") || true )
```

`diff` will return an error code if there is a difference causing the script to exit.

Comment (ericchiang) [reply scripts/verify-gopkg.sh]: it worked!

**Comment by ericchiang** [comment]
Gonna wait for https://github.com/kubernetes-incubator/bootkube/pull/877 to merge to push the change.

**Comment by rphillips** [comment]
merged #877

**Pull request merged**